src/DefaultBuilder/src/WebApplicationAuthenticationBuilder.cs

```diff
@@ -0,0 +1,45 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Authentication;
+
+internal class WebApplicationAuthenticationBuilder : AuthenticationBuilder
```

Comment:
Can we make this another PR, please? I already approved this one and it was already pretty big! I think it makes sense to ask for these together for preview5 approval, but another PR will make reviewing these much easier.

Reply:
Yeah, there's a secondary benefit to having these changes together since the config changes made the CLI are pointless without the framework changes to consume them.

The options-related changes are relatively small and can be viewed in this delta: https://github.com/dotnet/aspnetcore/pull/41520/commits/751c1d70cd89998b66369d7eb3149e6f229e41fe. The framework support doesn't require any changes to the existing CLI code.